stiebeleltron/integrationpluginstiebeleltron.cpp

```diff
@@ -87,36 +90,192 @@ void IntegrationPluginStiebelEltron::discoverThings(ThingDiscoveryInfo *info)
     });
 }
 
-void IntegrationPluginStiebelEltron::startMonitoringAutoThings()
-{
-
-}
+void IntegrationPluginStiebelEltron::startMonitoringAutoThings() {}
 
-void IntegrationPluginStiebelEltron::setupThing(ThingSetupInfo *info)
-{
+void IntegrationPluginStiebelEltron::setupThing(ThingSetupInfo *info) {
     Thing *thing = info->thing();
     qCDebug(dcStiebelEltron()) << "Setup" << thing << thing->params();
 
     if (thing->thingClassId() == stiebelEltronThingClassId) {
-
         QHostAddress address(thing->paramValue(stiebelEltronThingIpAddressParamTypeId).toString());
         quint16 port = thing->paramValue(stiebelEltronThingPortParamTypeId).toUInt();
         quint16 slaveId = thing->paramValue(stiebelEltronThingSlaveIdParamTypeId).toUInt();
 
-        StiebelEltronModbusConnection *connection = new StiebelEltronModbusConnection(address, port, slaveId, this);
+        StiebelEltronModbusConnection *connection =
+            new StiebelEltronModbusConnection(address, port, slaveId, this);
 
-        connection->connectDevice();
+        connect(connection, &StiebelEltronModbusConnection::connectionStateChanged, this,
```

Comment:
Just in case the thing object does not exist any more due to shutdown or removed thing, you could connect between `connection` and `thing`, that way you would prevent access to the thing object in the lambda when it might not exist any more:

    connect(connection, &StiebelEltronModbusConnection::connectionStateChanged, thing,

This would apply also to all the other connections in the `setupThing`.

Reply:
See https://github.com/nymea/nymea-plugins-modbus/pull/55/commits/a84be56d323e5081a1a22b8a6cf4af3998c68e7d